src/Sulu/Bundle/AdminBundle/Metadata/SchemaMetadata/PropertyMetadata.php

```diff
@@ -23,10 +23,16 @@ class PropertyMetadata
      */
     private $mandatory;
 
-    public function __construct(string $name, bool $mandatory)
+    /**
+     * @var array<string, mixed>|null
+     */
+    private $jsonSchema;
+
+    public function __construct(string $name, bool $mandatory, ?array $jsonSchema = null)
     {
         $this->name = $name;
         $this->mandatory = $mandatory;
+        $this->jsonSchema = $jsonSchema;
```

Comment:
Not sure what to think about this... I get that this is the fastest way of doing stuff, but we are mixing how these classes should be used. When following the current paradigm, then there would have to be a `ObjectMetadata` class along with the already existing `ArrayMetadata` and `ConstMetadata`, and all of them should inherit from the `PropertyMetadata`...

I think we could go for an approach where we are completely using arrays like in this property, but then we'd probably have to go for arrays completely. By that I mean no `SchemaMetadata` or `PropertyMetadata` at all, and handling all differently. A reason that would be problematic is that the [`SchemaMetadata` is holding some logic](https://github.com/sulu/sulu/blob/11fdd6d8de723374529bc0b3e4bfeecd1f6c28aa/src/Sulu/Bundle/AdminBundle/Metadata/SchemaMetadata/SchemaMetadata.php#L40-L85), which is actually quite nice.

So either approach would be fine for me, but mixing them does not feel like a good idea to me.

Reply:
I would go for option 1, but without renaming the `PropertyMetadataMapper` to `ObjectMetadataMapper`, right? Because it still maps a `PropertyMetadata` to an `ObjectMetadata`?

Reply:
I think I kind of answered that in https://github.com/sulu/sulu/pull/5640#discussion_r530859212